aws/sra-test/integration-tests/aws-sdk-s3/benches/middleware_vs_orchestrator.rs

```diff
@@ -0,0 +1,285 @@
+    use aws_sdk_s3::operation::list_objects_v2::ListObjectsV2Input;
+    use aws_smithy_client::erase::DynConnector;
+    use aws_smithy_runtime::client::connections::adapter::DynConnectorAdapter;
+    use aws_smithy_runtime_api::client::endpoints::StaticUriEndpointResolver;
+    use aws_smithy_runtime_api::client::interceptors::{
+        Interceptor, InterceptorContext, InterceptorError, Interceptors,
+    };
+    use aws_smithy_runtime_api::client::orchestrator::{
+        BoxError, ConfigBagAccessors, Connection, HttpRequest, HttpResponse, TraceProbe,
+    };
+    use aws_smithy_runtime_api::client::runtime_plugin::RuntimePlugin;
+    use aws_smithy_runtime_api::config_bag::ConfigBag;
+    use aws_types::region::SigningRegion;
+    use aws_types::SigningService;
+    use http::Uri;
+    use std::sync::Arc;
+
+    pub struct ManualServiceRuntimePlugin(pub DynConnector);
+
+    impl RuntimePlugin for ManualServiceRuntimePlugin {
+        fn configure(&self, cfg: &mut ConfigBag) -> Result<(), BoxError> {
+            let identity_resolvers =
+                aws_smithy_runtime_api::client::orchestrator::IdentityResolvers::builder()
+                    .identity_resolver(
+                        aws_runtime::auth::sigv4::SCHEME_ID,
+                        aws_runtime::identity::credentials::CredentialsIdentityResolver::new(
+                            SharedCredentialsCache::new(CredentialsCache::lazy().create_cache(
+                                SharedCredentialsProvider::new(Credentials::for_tests()),
+                            )),
+                        ),
+                    )
+                    .identity_resolver(
+                        "anonymous",
+                        aws_smithy_runtime_api::client::identity::AnonymousIdentityResolver::new(),
+                    )
+                    .build();
+            cfg.set_identity_resolvers(identity_resolvers);
+
+            let http_auth_schemes =
+                aws_smithy_runtime_api::client::orchestrator::HttpAuthSchemes::builder()
+                    .auth_scheme(
+                        aws_runtime::auth::sigv4::SCHEME_ID,
+                        aws_runtime::auth::sigv4::SigV4HttpAuthScheme::new(),
+                    )
+                    .build();
+            cfg.set_http_auth_schemes(http_auth_schemes);
+
+            cfg.set_auth_option_resolver(
+                aws_smithy_runtime_api::client::auth::option_resolver::AuthOptionListResolver::new(
+                    Vec::new(),
+                ),
+            );
+
+            //cfg.set_endpoint_resolver(DefaultEndpointResolver::new(
+            //    aws_smithy_http::endpoint::SharedEndpointResolver::new(
+            //        aws_sdk_s3::endpoint::DefaultResolver::new(),
+            //    ),
+            //));
+            cfg.set_endpoint_resolver(StaticUriEndpointResolver::uri(Uri::from_static(
+                "https://test-bucket.s3.us-east-1.amazonaws.com/",
+            )));
```

Comment:
This was where the endpoint JSON was getting reparsed for each request, which was causing the massive performance difference.

Reply:
Related to [the discussion](https://github.com/awslabs/smithy-rs/pull/2593/files#r1171662473), once we get the configuration working, we will no longer need to `aws_sdk_s3::endpoint::DefaultResolver::new()` within the `configure` method; `DefaultResolver` has already been created as part of the service config construction and been placed into the config bag. So the commented out code above will eventually look like the following:
```
cfg.set_endpoint_resolver(DefaultEndpointResolver::new(
    cfg.get::<SharedEndpointResolver>().expect("...").clone(),
));
```